source/common/stats/thread_local_store.cc

```diff
@@ -25,7 +25,10 @@ ThreadLocalStoreImpl::ThreadLocalStoreImpl(Allocator& alloc)
       tag_producer_(std::make_unique<TagProducerImpl>()),
       stats_matcher_(std::make_unique<StatsMatcherImpl>()), heap_allocator_(alloc.symbolTable()),
       null_counter_(alloc.symbolTable()), null_gauge_(alloc.symbolTable()),
-      null_histogram_(alloc.symbolTable()) {}
+      null_histogram_(alloc.symbolTable(), Histogram::Unit::Unspecified),
```

Comment:
do we need to have Null Histograms with the different units? Maybe we can just have null histograms always use unspecified units?

Reply:
We actually do, some histograms are used in `Timespans` and if you were to create a rejection rule to reject some of them, a `NullHistogram` of type `Unspecified` would get used in the `Timespan` and crash the process (ASSERT on the histogram type).

I wanted to reduce the ugliness by creating a `enum` -> `NullHistogram` map but realized the type is neither copyable nor movable so ended up just adding a couple of fields.

Technically speaking we can have only two - one `NullHistogram` for time and one for the rest but somehow it felt hacky (and no compiler warning on enum changes)?

Reply:
Fixed the way you suggested, pushing in a bit.